src/cmap/connection_pool.ts

```diff
@@ -544,7 +546,17 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
       fn(undefined, conn, (fnErr, result) => {
         if (typeof callback === 'function') {
           if (fnErr) {
-            callback(fnErr);
+            if ((fnErr as MongoError).code === MONGODB_ERROR_CODES.Reauthenticate) {
+              this.reauthenticate(conn, fn, (error, res) => {
+                if (error) {
+                  callback(error);
```

Comment:
I don't think this is quite right - when no callback is supplied, we don't attempt to re-authenticate?

We only use `withConnection` in server.command and a callback is always provided.  I think we should make `callback` required, which would make the guard on line 547 unnecessary and solve this issue (and remove a layer of nesting along the way 🙂).  thoughts?